### PR TITLE
enh(remote-server): Reduce exported objects and improve process duration

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

- Use cache to store services IDs list
- Use cache to store hosts IDs list
- Use cache to store SNMP Traps IDs list
- Remove meta-service for export (not available on Remote Server)
- Remote Downtimes for export (not available on Remote Server)
- Export SNMP Traps linked to templates of services

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Make an export with 3K hosts and 20K services, the duration was around 5min30s.
With patch the new duration is about 1min30.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

**Comment by lpinsivy** [comment]
@kduret done !

File: src/Centreon/Domain/Repository/CommandRepository.php [hunk 1]
```diff
@@ -121,7 +78,8 @@ public function exportList(array $list): array
 
         $sql = <<<SQL
 SELECT
-    t.*
+--    t.*
```
Comment (kduret) [inline src/Centreon/Domain/Repository/CommandRepository.php]: remove comments :)

File: src/Centreon/Domain/Repository/ServiceRepository.php [hunk 2]
```diff
@@ -162,4 +145,95 @@ public function getChainByParant($id, &$result)
...
+        $sql = <<<SQL
+SELECT l.* FROM (
+SELECT DISTINCT t.service_id AS `id`
+FROM service AS t
+INNER JOIN host_service_relation AS hsr ON hsr.service_service_id = t.service_id
+INNER JOIN ns_host_relation AS hr ON hr.host_host_id = hsr.host_host_id
+WHERE hr.nagios_server_id IN ({$ids})
+SQL;
+
+        // Extract BA services
```
Comment (kduret) [inline src/Centreon/Domain/Repository/ServiceRepository.php]: can you find a way to remove references to centreon-bam ?

File: src/CentreonRemote/Domain/Exporter/CommandExporter.php [hunk 3]
```diff
@@ -43,6 +43,7 @@ public function export(): void
             $this->_dump($command, $this->getFile(static::EXPORT_FILE_COMMAND));
         })();
 
+        /* Not necessary for Remote Server export
```
Comment (kduret) [inline src/CentreonRemote/Domain/Exporter/CommandExporter.php]: remove comments

File: src/CentreonRemote/Domain/Exporter/HostExporter.php [hunk 4]
```diff
@@ -37,84 +37,112 @@ public function export(): void
...
+        (function () use ($hostList, $hostTemplateChain) {
             $hostGroupRelation = $this->db
                 ->getRepository(Repository\HostGroupRelationRepository::class)
-                ->export($pollerIds, $hostTemplateChain)
+                ->export($hostList, $hostTemplateChain)
             ;
             $this->_dump($hostGroupRelation, $this->getFile(static::EXPORT_FILE_GROUP_RELATION));
         })();
 
+        /* Not necessary for Remote Server export
```
Comment (kduret) [inline src/CentreonRemote/Domain/Exporter/HostExporter.php]: remove comments

File: src/CentreonRemote/Domain/Exporter/HostExporter.php [hunk 5]
```diff
@@ -180,6 +208,7 @@ public function import(): void
             }
         })();
 
+        /* Not necessary for Remote Server export
```
Comment (kduret) [inline src/CentreonRemote/Domain/Exporter/HostExporter.php]: remove comments

File: src/CentreonRemote/Domain/Exporter/PollerExporter.php [hunk 6]
```diff
@@ -67,13 +67,15 @@ public function export(): void
             $this->_dump($data, $this->getFile(static::EXPORT_FILE_CFG_RESOURCE_INSTANCE_RELATION));
         })();
 
+        /* Not necessary for Remote Server export
```
Comment (kduret) [inline src/CentreonRemote/Domain/Exporter/PollerExporter.php]: remove comments

File: src/CentreonRemote/Domain/Exporter/PollerExporter.php [hunk 7]
```diff
@@ -107,13 +109,15 @@ public function export(): void
             $this->_dump($cfgCentreonBrokerInfo, $this->getFile(static::EXPORT_FILE_CFG_CENTREONBROKER_INFO));
         })();
 
+        /* Not necessary for Remote Server export
```
Comment (kduret) [inline src/CentreonRemote/Domain/Exporter/PollerExporter.php]: remove comments

File: src/CentreonRemote/Domain/Exporter/ServiceExporter.php [hunk 8]
```diff
@@ -36,77 +36,104 @@ public function export(): void
...
             $serviceCategoryRelation = $this->db
                 ->getRepository(Repository\ServiceCategoryRelationRepository::class)
-                ->export($pollerIds, $serviceTemplateChain)
+                ->export($serviceList, $serviceTemplateChain)
             ;
             $this->_dump($serviceCategoryRelation, $this->getFile(static::EXPORT_FILE_CATEGORY_RELATION));
         })();
 
-        (function () use ($pollerIds, $serviceTemplateChain) {
+        // Extract on demande macros data used by services
```
Comment (kduret) [inline src/CentreonRemote/Domain/Exporter/ServiceExporter.php]:
```suggestion
        // Extract on demand macros data used by services
```

File: src/Centreon/Domain/Repository/ServiceRepository.php (hunk 2 above)
Comment (lpinsivy) [reply src/Centreon/Domain/Repository/ServiceRepository.php]: Could be impossible with actual system without rewrite some part.
May be for latter

**Comment by kduret** [comment]
replaced by #7749

**Pull request closed without merging**